internal/config/supervisor/config_test.go

```diff
@@ -208,3 +321,74 @@ func TestFromPath(t *testing.T) {
 		})
 	}
 }
+
+func TestAddrIsOnlyOnLoopback(t *testing.T) {
+	tests := []struct {
+		addr string
+		want bool
+	}{
+		{addr: "localhost:", want: true},
+		{addr: "localhost:0", want: true},
+		{addr: "localhost:80", want: true},
+		{addr: "localhost:http", want: true},
+		{addr: "ip6-localhost:", want: true},
+		{addr: "ip6-localhost:0", want: true},
+		{addr: "ip6-localhost:80", want: true},
+		{addr: "ip6-localhost:http", want: true},
+		{addr: "ip6-loopback:", want: true},
+		{addr: "ip6-loopback:0", want: true},
+		{addr: "ip6-loopback:80", want: true},
+		{addr: "ip6-loopback:http", want: true},
+		{addr: "127.0.0.1:", want: true},
+		{addr: "127.0.0.1:0", want: true},
+		{addr: "127.0.0.1:80", want: true},
+		{addr: "127.0.0.1:http", want: true},
+		{addr: "[::1]:", want: true},
+		{addr: "[::1]:0", want: true},
+		{addr: "[::1]:80", want: true},
+		{addr: "[::1]:http", want: true},
+		{addr: "[0:0:0:0:0:0:0:1]:", want: true},
+		{addr: "[0:0:0:0:0:0:0:1]:0", want: true},
+		{addr: "[0:0:0:0:0:0:0:1]:80", want: true},
+		{addr: "[0:0:0:0:0:0:0:1]:http", want: true},
+		{addr: "", want: false},               // illegal input, can't be empty
+		{addr: "host", want: false},           // illegal input, need colon
+		{addr: "localhost", want: false},      // illegal input, need colon
+		{addr: "127.0.0.1", want: false},      // illegal input, need colon
+		{addr: ":", want: false},              // illegal input, need either host or port
+		{addr: "2001:db8::1:80", want: false}, // illegal input, forgot square brackets
+		{addr: ":0", want: false},
+		{addr: ":80", want: false},
+		{addr: ":http", want: false},
+		{addr: "notlocalhost:", want: false},
+		{addr: "notlocalhost:0", want: false},
+		{addr: "notlocalhost:80", want: false},
+		{addr: "notlocalhost:http", want: false},
+		{addr: "0.0.0.0:", want: false},
+		{addr: "0.0.0.0:0", want: false},
+		{addr: "0.0.0.0:80", want: false},
+		{addr: "0.0.0.0:http", want: false},
+		{addr: "[::]:", want: false},
+		{addr: "[::]:0", want: false},
+		{addr: "[::]:80", want: false},
+		{addr: "[::]:http", want: false},
+		{addr: "42.42.42.42:", want: false},
+		{addr: "42.42.42.42:0", want: false},
+		{addr: "42.42.42.42:80", want: false},
+		{addr: "42.42.42.42:http", want: false},
+		{addr: "[2001:db8::1]:", want: false},
+		{addr: "[2001:db8::1]:0", want: false},
+		{addr: "[2001:db8::1]:80", want: false},
+		{addr: "[2001:db8::1]:http", want: false},
+		{addr: "[fe80::1%zone]:", want: false},
+		{addr: "[fe80::1%zone]:0", want: false},
+		{addr: "[fe80::1%zone]:80", want: false},
+		{addr: "[fe80::1%zone]:http", want: false},
+	}
+	for _, test := range tests {
+		tt := test
+		t.Run(fmt.Sprintf("address %s should be %t", tt.addr, tt.want), func(t *testing.T) {
+			require.Equal(t, tt.want, addrIsOnlyOnLoopback(tt.addr))
```

Comment:
```suggestion
		t.Run(fmt.Sprintf("address %s should be %t", tt.addr, tt.want), func(t *testing.T) {
			t.Parallel()
			require.Equal(t, tt.want, addrIsOnlyOnLoopback(tt.addr))
```

Reply:
Updated in 51c527a965a743f89347420e6a5fe075a59ef3f0.